### PR TITLE
feature/COR-1758_shorten-sidebar-with-hooks

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard30**
Refactored the sidebar to make use of hooks

### CODE REVIEW

File: packages/common/src/data/reverse-router.ts [hunk 1]
```diff
@@ -49,3 +49,32 @@ export function getReverseRouter(isMobile: boolean) {
...
   return reverseRouter;
 }
+
+export function getArchivedRoutes() {
+  /**
+   * Add the same keys that are part of the archived_metrics array
+   * in the nl-layout and gm-layout files
+   */
+
+  const archivedPaths = {
```
Comment (VWSCoronaDashboard29) [inline packages/common/src/data/reverse-router.ts:59]: I think this is an OK way to solve the issue of archived paths for now. I'd suggest that in the future we can look into a way of splitting the NL paths block on line 17 into archived and non-archived paths and work from there.

**Review by VWSCoronaDashboard29 — CHANGES_REQUESTED** [review]
Looks quite allright to me but what about the cijferverantwoording?

File: packages/app/src/components/aside/menu.tsx [hunk 2]
```diff
@@ -99,6 +135,21 @@ function isActivePath(router: NextRouter, href: Url) {
   return currentPath === hrefPath;
 }
 
+function isCategoryMenuOpen(router: NextRouter, archivedPaths: ArchivedPath) {
```
Comment (VWSCoronaDashboard29) [inline packages/app/src/components/aside/menu.tsx:138]: I would suggest the following name.

```suggestion
function isCurrentRouteArchivedPage(router: NextRouter, archivedPaths: ArchivedPath) {
```

**Review by VWSCoronaDashboard29 — DISMISSED** [review]

File: packages/app/src/components/aside/menu.tsx (hunk 2 above)
Comment (VWSCoronaDashboard30) [reply packages/app/src/components/aside/menu.tsx]: The function name was changed!

File: packages/common/src/data/reverse-router.ts (hunk 1 above)
Comment (VWSCoronaDashboard30) [reply packages/common/src/data/reverse-router.ts:59]: True, we should look into this, as we also have the third sidebar

**Review by VWSCoronaDashboard29 — APPROVED** [review]

**Pull request merged**